src/mem/metaslab.h

```diff
@@ -208,18 +210,34 @@ namespace snmalloc
      */
     uintptr_t get_remote_and_sizeclass()
     {
-      return remote_and_sizeclass;
+      return remote_and_sizeclass.load();
     }
 
     [[nodiscard]] RemoteAllocator* get_remote() const
     {
       return reinterpret_cast<RemoteAllocator*>(
-        pointer_align_down<alignof(RemoteAllocator)>(remote_and_sizeclass));
+        pointer_align_down<alignof(RemoteAllocator)>(
+          remote_and_sizeclass.load()));
     }
 
     [[nodiscard]] sizeclass_t get_sizeclass() const
     {
-      return remote_and_sizeclass & (alignof(RemoteAllocator) - 1);
+      return static_cast<sizeclass_t>(remote_and_sizeclass.load()) &
+        (alignof(RemoteAllocator) - 1);
+    }
+
+    /**
+     * Attempts to atomically take ownership of a chunk that is being
+     * deallocted.  May fail if multiple threads attempt at once, but spurious
+     * failures are not permitted; one thread must win in the event of a race.
+     */
+    [[nodiscard]] ChunkRecord* claim_for_large_dealloc(sizeclass_t sizeclass)
```

Comment:
I would prefer this to be part of `backend.h`.  This is something that a new backend has to support, and if the pagemap is mapped read-only then that could be complex to achieve.

Reply:
+1